.github/workflows/qc.yml

```diff
@@ -9,6 +9,8 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  pull_request_target:
```

Comment:
From my understanding of [the documentation on the `pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target), this is not suitable:

> This event runs in the context of the base of the pull request, rather than in the context of the merge commit

This means that we check out the repository to run the test suite, the commit that will be checked out is the **base** from which the foreign branch was forked, _not_ the **tip** of the foreign branch. So the test suite will be triggered, but it will only test contents that were already present in the repository, **without** the changes that the PR is proposing to introduce!

Reply:
Yes, but there's this sentence here.

> This event allows your workflow to do things like label or comment on pull requests from forks.

Reply:
But what would be the point of allowing to comment if the test suite runs against the base of the PR? It wouldn’t test anything useful – none of the changes from the PR would be tested.

Reply:
Ok, only need to give permission then.